src/dlgTriggerEditor.cpp

```diff
@@ -7324,7 +7324,9 @@ void dlgTriggerEditor::slot_changeEditorTextOptions( QTextOption::Flags state )
 void dlgTriggerEditor::clearDocument(edbee::TextEditorWidget* ew, const QString& initialText) {
 
     // This will cause a crash on the next redoClear because of an unregistered view and a >0 redo count
-    mpSourceEditorEdbeeDocument = newTextDocument();
+
+    mpSourceEditorEdbeeDocument = new edbee::CharTextDocument();
+    mpSourceEditorEdbeeDocument->setLanguageGrammar(edbee::Edbee::instance()->grammarManager()->detectGrammarWithFilename(QLatin1Literal("Buck.lua")));
```

Comment:
When I first started working on implementing edbee, I had no luck applying the lua lexer directly; I could only get it to work by using it the same way edbee-app did it: referencing it through detect by filename.